src/lib/zod/index.ts

```diff
@@ -94,6 +97,36 @@ export const deleteEpochInput = z
   })
   .strict();
 
+export const createEpochInput = z
+  .object({
+    circle_id: z.number().int().positive(),
+    start_date: zStringISODateUTC,
+    repeat: z.number().int().min(0).max(2),
+    days: z
+      .number()
+      .min(1, 'Must be at least one day.')
+      .max(100, 'cant be more than 100 days'),
+    grant: z.number().positive().min(1).max(1000000000).optional(),
+  })
+  .strict()
+  .superRefine((val, ctx) => {
+    let message;
+    if (val.days > 7 && val.repeat === 1) {
+      message =
+        'You cannot have more than 7 days length for a weekly repeating epoch.';
+    } else if (val.days > 28 && val.repeat === 2) {
+      message =
+        'You cannot have more than 28 days length for a monthly repeating epoch.';
+    }
```

Comment:
Is this a departure from the incumbent logic in laravel? I thought we could have epoch lengths that were greater than 28 days but would be truncated down to a shorter month's length if necessary.

Reply:
i copied the validation over from what i did in the past
https://github.com/coordinape/coordinape-backend/blob/main/app/Http/Requests/NewEpochRequest.php#L41
 
there is also a logic that trims down the start day for repeating epochs
e.g for monthly if start date is 30th when it comes to repeating in Feburary the start date will be trimmed to 28th

https://github.com/coordinape/coordinape-backend/blob/main/app/Repositories/EpochRepository.php#L90

Reply:
> there is also a logic that trims down the start day for repeating epochs
e.g for monthly if start date is 30th when it comes to repeating in Feburary the start date will be trimmed to 28th

I see. Is this legacy functionality then for existing circles? It'd be nice to remove it, or make it consistent with the current way we create epochs. We should probably just wait to create the new schema to sort this out. Thanks for the explanation!

Reply:
> > there is also a logic that trims down the start day for repeating epochs
> > e.g for monthly if start date is 30th when it comes to repeating in Feburary the start date will be trimmed to 28th
> 
> I see. Is this legacy functionality then for existing circles? It'd be nice to remove it, or make it consistent with the current way we create epochs. We should probably just wait to create the new schema to sort this out. Thanks for the explanation!

How would you truncate the end dates to make it not overlap with the subsequent repeating epochs ? Do you mean creating the repeating epochs in advance ?